fabtests/ubertest/uber.c

```diff
@@ -652,82 +480,125 @@ static int ft_fw_server(void)
 
 		results[ft_fw_result_index(ret)]++;
 
-	} while (!ret || ret == FI_EIO || ret == FI_ENODATA);
+	} while (!ret || ret == -FI_EIO || ret == -FI_ENODATA || ret == -FT_SKIP);
 
 	return ret;
 }
-
-static int ft_client_child(void)
+static int ft_client_setup(struct fi_info *hints, struct fi_info *info)
 {
-	struct fi_info *hints, *info;
-	int ret, result, server_ready = 0;
+	int ret = 0;
+	ret = ft_recv_test_info();
+	if (ret) {
+		FT_PRINTERR("ft_recv_test_info", ret);
+		return ret;
```

Comment:
Why isn't ft_send_result() called if this error occurs, similar to how it is called below?

Reply:
I think the idea was is that if this fails its already a sockets error, so attempting to do another socket communication would be optimistic at best.